cmd/test/test.go

```diff
@@ -0,0 +1,165 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package test
+
+// See also the Conftest version of this command here:
+// https://github.com/open-policy-agent/conftest/blob/master/internal/commands/test.go
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	hd "github.com/MakeNowJust/heredoc"
+	"github.com/open-policy-agent/conftest/output"
+	"github.com/open-policy-agent/conftest/runner"
+	"github.com/spf13/cobra"
+
+	"github.com/enterprise-contract/ec-cli/internal/applicationsnapshot"
+)
+
+const (
+	OutputAppstudio = "appstudio"
+)
+
+var TestCmd *cobra.Command
+
+func fetchTestCmd() *cobra.Command {
+	flags := struct {
+		Policy             []string
+		Data               []string
+		Namespace          []string
+		AllNamespaces      bool
+		NoFail             bool
+		FailOnWarn         bool
+		Output             string
+		NoColor            bool
+		SuppressExceptions bool
+		Trace              bool
+		JunitHideMessage   bool
+	}{}
+
+	cmd := &cobra.Command{
+		Use:   "test [flags] input",
+		Short: "A wrapper for the conftest test command",
+
+		Long: hd.Doc(`
+			The ec test command is meant to work as a drop-in replacement for conftest test.
+
+			It supports most but not all of the flags supported by Conftest. Its supports one
+			more output format than Conftest, the 'appstudio' format, which is suitable for use
+			with the standard AppStudio TEST_OUTPUT Tekton result format.
+		`),
+
+		Example: hd.Doc(`
+			The standard usage is similar to Conftest:
+
+				ec test --policy my_policies.rego --data my_data.json --namespace main input.json
+
+			Like Conftest it will traverse directories to find multiple policies and data files:
+
+				ec test --policy ./policy --data ./data --namespace main input.json
+		`),
+
+		RunE: func(cmd *cobra.Command, fileList []string) (err error) {
+			if len(fileList) < 1 {
+				return fmt.Errorf("at least one input file must be specified")
+			}
+
+			r := &runner.TestRunner{
+				Policy:             flags.Policy,
+				Data:               flags.Data,
+				Namespace:          flags.Namespace,
+				AllNamespaces:      flags.AllNamespaces,
+				NoFail:             flags.NoFail,
+				FailOnWarn:         flags.FailOnWarn,
+				Output:             flags.Output,
+				NoColor:            flags.NoColor,
+				SuppressExceptions: flags.SuppressExceptions,
+				Trace:              flags.Trace,
+			}
+
+			ctx := cmd.Context()
+			results, err := r.Run(ctx, fileList)
+			if err != nil {
+				return fmt.Errorf("running test: %w", err)
+			}
+
+			if flags.Output == OutputAppstudio {
+				// The appstudio format is unknown to Conftest so we handle it ourselves
+				report := applicationsnapshot.AppstudioReportFromCheckResults(results)
+				reportOutput, err := json.Marshal(report)
+				if err != nil {
+					return fmt.Errorf("output results: %w", err)
+				}
+				fmt.Printf("%s", reportOutput)
```

Comment:
For easier testing, consider using `cmd.OutOrStdout()`.

Reply:
Done.

Reply:
Beware I don't think we can send the standard conftest output to anything other than os.Stdout.